### DISCUSSION TITLE
shortcuts

### DISCUSSION THREAD

**Discussion by genbread** [bnpr/Malt#301] [Feedback & Feature Requests]
It would be helpful to have shortcuts like with Node Wrangler in the malt nodes editor. It's annoying to have to drag all the way across the editor to the output whenever I want to preview a section of the nodes. I also find it quite tedious, constantly going through the massive dropdown of the functions and then other massive dropdowns past it for every little node. Maybe have something like the math node in Eevee, where all those functions can be accessed from one node, perhaps a master node for each of the vecx, float, and int functions?

**Comment by pragma37** [comment]
What I usually do is type the node name in the operator search menu (F3).

> perhaps a master node for each of the vecx, float, and int functions?

Yes, this is something I have thought about adding. Being able to mark some glsl files to work that way.

Node Wrangler-like shortcuts would be tricky to implement in a clean way, given how Malt works, since almost all the nodes are generated by the active pipeline so the shortcuts would have to be pipeline specific.

Maybe this would be a good fit for the addon @Kolupsy is working on ? (https://github.com/bnpr/Malt/discussions/240#discussioncomment-2318296)

**Reply by Kolupsy** [reply]
@pragma37  The improved/more user-friendly node categories are under way, I have thought about some equivalent to the preview function of node wrangler, maybe you could flag IO nodes to be the target of previewing your nodes or something. But before that can be released we of course need to fix the crashing issue in #293 .
